invenio_cli/helpers/commands.py

```diff
@@ -157,10 +162,14 @@ def services(self, force):
 
         NOTE: We use check=True to mimic set -e from original setup script
               i.e. if a command fails, an exception is raised
+
+        A check is done to see if one-time setup has been executed before
+        and confirmation saved in invenio-cli config file.
         """
         self._ensure_containers_running()
 
         if force:
+            self.cli_config.update_services_setup(False)
```

Comment:
moderate: I think having this set is a really good idea. Nonetheless, I see a potential issue:

a) It's marked as set, an error occurs. Half are set half are not.
b) It's marked as not set, an error occurs. Half are set half are not.

In both cases retrying would case error in some of them... (would happen anyway if we dont have this flag, so having it is already an improvement to avoid setting if correct. So I think it would be an issue for the future (i.e. to be treated separately)

Conclusion: Should we run this after the setup commands? So we only have "set" if it's for sure set correctly? I think is better to have an error for "Some were set but it fails" rather than "They didnt got set but it said it is". I hope I explained myself correct, sorry if it's confusing :sweat_smile:

Reply:
I think I understand and I thought I had this covered, but you are right that if an error occurs in one of those forced commands, we get into a situation where `invenio-cli services` will keep raising an error: (concrete example just so we are on the same page)

Let's say the `invenio index destroy` underlying call raises an exception when running `invenio-cli services --force`, running `invenio-cli services --force` again will break, because now `invenio db destroy --yes-i-know` doesn't find the database! (it destroyed it last time around). `invenio-cli services` will break too for that matter since `invenio index init`  will complain we are trying to create a new one with the same name!

So wherever we put the set code, we still get a problem. Putting it at the end of the force block is a little better perhaps - moved it :+1:   

To solve our problems, we need an idempotent version of all those commands : 
- invenio db destroy: https://github.com/inveniosoftware/invenio-db/issues/126
- invenio index destroy: https://github.com/inveniosoftware/invenio-indexer/issues/114
- invenio index queue init purge : same as above
(and the regular ones too)

Reply:
Fully agree, if we get them idempotent it should be fine with it at the end. Will push for those to get solved. Thanks!